opflexagent/vif_plug_vpp/vpp.py

```diff
@@ -0,0 +1,153 @@
+                   default=120,
+                   help='Amount of time, in seconds, that vpp_api should '
+                   'wait for a response from the database. 0 is to wait '
+                   'forever.',
+                   deprecated_group="DEFAULT"),
+    )
+
+    @staticmethod
+    def gen_port_name(prefix, id):
+        return ("%s%s" % (prefix, id))[:VppPlugin.NIC_NAME_LEN]
+
+    def describe(self):
+        pp_ovs = objects.host_info.HostPortProfileInfo(
+            profile_object_name=
+            objects.vif.VIFPortProfileOpenVSwitch.__name__,
+            min_version="1.0",
+            max_version="1.0",
+        )
+        pp_ovs_representor = objects.host_info.HostPortProfileInfo(
+            profile_object_name=
+            objects.vif.VIFPortProfileOVSRepresentor.__name__,
+            min_version="1.0",
+            max_version="1.0",
+        )
+        return objects.host_info.HostPluginInfo(
+            plugin_name=constants.PLUGIN_NAME,
+            vif_info=[
+                objects.host_info.HostVIFInfo(
+                    vif_object_name=objects.vif.VIFBridge.__name__,
+                    min_version="1.0",
+                    max_version="1.0",
+                    supported_port_profiles=[pp_ovs]),
+                objects.host_info.HostVIFInfo(
+                    vif_object_name=objects.vif.VIFOpenVSwitch.__name__,
+                    min_version="1.0",
+                    max_version="1.0",
+                    supported_port_profiles=[pp_ovs]),
+                objects.host_info.HostVIFInfo(
+                    vif_object_name=objects.vif.VIFVHostUser.__name__,
+                    min_version="1.0",
+                    max_version="1.0",
+                    supported_port_profiles=[pp_ovs, pp_ovs_representor]),
+                objects.host_info.HostVIFInfo(
+                    vif_object_name=objects.vif.VIFHostDevice.__name__,
+                    min_version="1.0",
+                    max_version="1.0",
+                    supported_port_profiles=[pp_ovs, pp_ovs_representor]),
+            ])
+
+    def _get_mtu(self, vif):
+        if vif.network and vif.network.mtu:
+            return vif.network.mtu
+        return self.config.network_device_mtu
+
+    def _create_vif_port(self, vif, vif_name, instance_info, **kwargs):
+        mtu = self._get_mtu(vif)
+        linux_net.create_vpp_vif_port(
+            vif_name,
+            vif.port_profile.interface_id,
+            vif.address, instance_info.uuid,
+            mtu,
+            timeout=self.config.vpp_api_timeout,
+            **kwargs)
+
+    def _update_vif_port(self, vif, vif_name):
+        mtu = self._get_mtu(vif)
+        linux_net.update_vpp_vif_port(vif_name, mtu)
+
+    def _plug_vhostuser(self, vif, instance_info):
+        vif_name = VppPlugin.gen_port_name(
+            constants.VPP_VHOSTUSER_PREFIX, vif.id)
+        args = {}
+        if vif.mode == "client":
+            args['interface_type'] = \
```

Comment:
please use brackets instead of slash for line continuation

Reply:
ok